kongswap_adaptor/tests/lifecycle.rs

```diff
@@ -0,0 +1,712 @@
+        owner: *SNS_GOVERNANCE_CANISTER_ID,
+        subaccount: Some(compute_treasury_subaccount_bytes(
+            *SNS_GOVERNANCE_CANISTER_ID,
+        )),
+    };
+
+    let original_wasm = get_kong_adaptor_wasm();
+
+    // Phase I: setup the canister
+    // When we freshly deploy the pool, the desired amounts of both
+    // tokens are moved to the pool. Therefore, we expect
+    // SNS:
+    //      reserve_sns = 100 * E8 - 2 * FEE_SNS
+    // ICP:
+    //      reserve_icp = 100 * E8 - 2 * FEE_ICP
+    setup_kongswap_adaptor(
+        &mut agent,
+        sns_ledger_canister_id,
+        icp_ledger_canister_id,
+        kong_adaptor_canister_id,
+        &original_wasm,
+        treasury_icp_account,
+        treasury_sns_account,
+        100 * E8,
+    )
+    .await;
+
+    // Phase II: another deposit
+    // As the second deposit is proportional to the existing reserves
+    // of the pool, it gets thoroughly transferred to the pool
+    // SNS:
+    //      reserve_sns += 10 - 2 * FEE_SNS
+    // ICP:
+    //      reserve_icp += 10 - 2 * FEE_ICP
+    deposit(
+        &mut agent,
+        sns_ledger_canister_id,
+        icp_ledger_canister_id,
+        kong_adaptor_canister_id,
+        treasury_sns_account,
+        treasury_icp_account,
+        10 * E8,
+    )
+    .await;
+    // Phase III: first trade
+    // Here, we try to swap 1 * E8 of SNS and receive ICP in return.
+    // The amount of ICP withdrawn from the kongswap backend is
+    // amount_icp = amount_sns * reserve_icp / (reserve_sns + amount_sns)
+    // where amount_sns = 1 * E8 - 2 * FEE_SNS
+    // SNS:
+    //      reserve_sns += amount_sns
+    // ICP:
+    //      reserve_icp -= amount_icp
+    //      lp_fee_1 = 30 * amount_icp / 10_000
+    trade(
+        &mut agent,
+        sns_ledger_canister_id,
+        icp_ledger_canister_id,
+        1 * E8,
+        true,
+    )
+    .await;
+    // Phase IV: another deposit
+    // Now, as the pool is no longer in 1:1 price point, and we have more SNS
+    // some amount of ICP will be returned to the user (one more fee)
+    // The amount of icp to be moved into the pool is calculated as
+    // amount_icp = (reserve_icp / reserve_sns) * deposit_amount
+    // where depoit_amount = 10 * E8 - 2 * FEE_SNS
+    // SNS:
+    //      reserve_0 += deposit_amount
+    // ICP:
+    //      reserve_1 += amount_icp calculated above
+    deposit(
+        &mut agent,
+        sns_ledger_canister_id,
+        icp_ledger_canister_id,
+        kong_adaptor_canister_id,
+        treasury_sns_account,
+        treasury_icp_account,
+        10 * E8,
+    )
+    .await;
+    // Phase V: second trade
+    // Here, we try to swap 1 * E8 of ICP and receive SNS in return.
+    // The amount of SNS withdrawn from the kongswap backend is
+    // amount_sns = amount_icp * reserve_sns / (reserve_icp + amount_icp)
+    // where amount_icp = 1 * E8 - 2 * FEE_ICP
+    // SNS:
+    //      reserve_sns= 121 * E8 - 8 * FEE_SNS - amount_sns
```

Comment:
The number 121 doesn't appear anywhere else explicitly, consider rephrasing this comment to make it clear where the values come from.